src/stream/src/task/barrier_manager/managed_state.rs

```diff
@@ -222,14 +255,19 @@ impl ManagedBarrierState {
                     },
                 ..
             }) => {
-                let remaining_actors = actor_ids_to_collect
+                let remaining_actors: HashSet<ActorId> = actor_ids_to_collect
                     .into_iter()
                     .filter(|a| !collected_actors.remove(a))
                     .collect();
+                for actor_id in &self.failure_actors {
+                    if remaining_actors.contains(actor_id) {
+                        bail!("Actor {} exit unexpectedly", actor_id);
```

Comment:
Can we also throw the original error here?

Reply:
Well, the `StreamError` does not implement `Clone` trait, and it seems quite hard to do that. 😢  Do you think storing an internal error with original error encoded in `failure_actors` is acceptable?

Reply:
Seems there's no need to `Clone` the error. We can take ownership of the error instead of `inspect` it. 🤩